src/Orleans.Core/Diagnostics/ActivityPropagationGrainCallFilter.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime
+{
+    internal static class ActivityPropagationGrainCallFilter
```

Comment:
Could we make this an abstract class instead and have the others inherit from it rather than be nested?